stripe/src/main/java/com/stripe/android/model/Card.java

```diff
@@ -211,6 +280,64 @@ public Card build() {
     }
 
     /**
+     * Create a Card object from a raw JSON string.
+     *
+     * @param jsonString the JSON string representing the potential Card
+     * @return A Card if one can be made from the JSON, or {@code null} if one cannot be made
+     * or the JSON is invalid.
+     */
+    @Nullable
+    public static Card fromString(String jsonString) {
+        try {
+            JSONObject jsonObject = new JSONObject(jsonString);
+            return fromJson(jsonObject);
+        } catch (JSONException ignored) {
+            return null;
+        }
+    }
+
+    @Nullable
+    public static Card fromJson(JSONObject jsonObject) {
+        if (jsonObject == null || !VALUE_CARD.equals(jsonObject.optString(FIELD_OBJECT))) {
+            return null;
+        }
+
+        Integer expMonth = optInteger(jsonObject, FIELD_EXP_MONTH);
+        Integer expYear = optInteger(jsonObject, FIELD_EXP_YEAR);
+
+        //It's okay for the month to be missing, but not for it to be outside 1-12
+        if (expMonth != null && (expMonth < 1 || expMonth > 12)) {
+            return null;
+        }
+
+        if (expYear != null && expYear < 0) {
+            return null;
+        }
```

Comment:
I'm only checking to make sure the data is not invalid (like a month of 13), since that would create a card that makes no sense. There are other cases where I check for validity (brand name and funding type), though in both of those cases I don't actually return a null card, I just put in null values for those attributes. Perhaps I should do the same for the year and month.